src/linear_algebra/direct_sum_module.lean

```diff
@@ -6,6 +6,7 @@ Authors: Kenny Lau
 Direct sum of modules over commutative rings, indexed by a discrete type.
 -/
 import algebra.direct_sum
+import algebra.internal_direct_sum
```

Comment:
@kbuzzard, are you likely to want a new `internal_direct_sum_module` file too? Or should we flip the import direction to put `direct_sum.submodule_is_internal` inside `algebra/internal_direct_sum` too?

Reply:
I'm afraid I have very little clue about things like imports. All I'd like to do is to get some of that code which you and I wrote a couple of months ago for LTE (e.g. https://github.com/leanprover-community/lean-liquid/blob/cd0a8fd020876c98c8c592965bd8a9c790ff6f79/src/for_mathlib/grading.lean ) which ended up not being used, into mathlib. I thought that this PR might be a place to start but I hadn't thought about what should be going where. If it's any help, a concrete goal is the following. Let R be a commutative ring, let A be an R-algebra internally graded by sub-R-modules indexed by a type I. Then I want an API for the projection maps proj i : A ->_l[R] A (i.e. proj i \circ proj j = 0 if i != j, proj i is idempotent, a is sum over i of proj i a etc). What do you recommend?

Reply:
OK so I've closed this. All I want is a way of saying "This R-module A is isomorphic to the direct sum of these submodules A_i" and "here is the R-module projection map from A to A_i, modelled as an R-module map from A to A with range equal to A_i". I don't care about groups or monoids or anything, and it's not clear to me *in practice* whether we will need to make an API for these more primitive things, although of course we can still do it in theory.